src/engineChange.js

```diff
@@ -57,15 +57,24 @@ function loadStoredEngine() {
     var engine = EngineVersions[engineId];
     var queryVariable = getQueryVariable("engine_" + engineId);
 
-    if (queryVariable && engine.versions[queryVariable] !== undefined) {
-      window.localStorage["engine_" + engineId] = queryVariable;
-      window.location.href = window.location.search.replace("engine_" + engineId + "=" + queryVariable, "");
-    } else if (window.localStorage["engine_" + engineId] === undefined) {
+    // if engine version is missing, add the default value from dict 'EngineVersions' to the url:
```

Comment:
How about if no "engine_" URL is provided, then just use default version and do not set these URL parameters?

I believe that would be the simplest scenario from user POV - it works by default and URL parameters are used only if its required.

Reply:
Parameters are not visible in the URL now, when default parameters are chosen.